components/hab/src/command/plan/render.rs

```diff
@@ -0,0 +1,176 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs::create_dir_all;
+use std::fs::{File, read_to_string};
+use std::io::{Write};
+use std::path::Path;
+use serde_json::{self, Value as Json};
+use toml::Value;
+
+use crate::common::templating::TemplateRenderer;
+use crate::common::ui::{Status, UIWriter, UI};
+use crate::error::Result;
+
+pub fn start(
+    ui: &mut UI,
+    template_path: String,
+    default_toml_path: String,
+    user_toml_path: Option<String>,
+    mock_data_path: Option<String>,
+    print: bool,
+    no_render_dir: bool,
+    render_dir: String,
+    quiet: bool,
+) -> Result<()> {
+    // Strip the file name out of our passed template
+    let file_name = match Path::new(&template_path).file_name() {
+        Some(name) => name.to_str().clone().unwrap(),
+        None => panic!(format!("Something went wrong getting filename of {}", &template_path)),
+    }; 
+
+    if !(quiet) {
+      ui.begin(format!("Rendering: {} into: {}/ as: {}", template_path, render_dir, file_name))?;
+      ui.br()?;
+    }
+
+    // read our template from file
+    let template = read_to_string(&template_path)
+        .expect(&format!("something went wrong reading: {}", template_path)); 
+
+    // create a "data" json struct
+    let mut data: Json = serde_json::from_str("{}").unwrap();
+
+    if !(quiet) {
+        // import default.toml values, convert to JSON
+        ui.begin(format!("Importing default.toml: {}", &default_toml_path))?;
+    }
+
+    // we should always have a default.toml, would be nice to "autodiscover" based on package name,
+    // for now assume we're working in the plan dir if --default-toml not passed
+    let default_toml = read_to_string(&default_toml_path)
+        .expect(&format!("Something went wrong reading: {}", &default_toml_path));
+    // merge default into data struct
+    merge(&mut data, toml_to_json(&default_toml));
+
+    // import default.toml values, convert to JSON
+    // ui.begin(format!("Importing user.toml: {}", &user_toml_path));
+    let user_toml = match user_toml_path {
+        Some(path) => {
+            if !(quiet) {
+              // print helper message, maybe only print if '--verbose'? how?
+              ui.begin(format!("Importing user.toml: {}", path.to_string()))?;
+            }
+            read_to_string(path.to_string())
+                .expect(&format!("Something went wrong reading: {}", path.to_string()))
```

Comment:
```suggestion
                ?
```
Again, we should propagate the error rather than panicking.